dicomImport/matRad_importFieldShapes.m

```diff
@@ -80,54 +80,66 @@
         
        if isfield(currControlPointElement, 'BeamLimitingDevicePositionSequence')
            % get the leaf position for every device
-           collimation.Fields(counter).LeafPos{length(currDeviceSeqNames),1} = [];
+           tmpCollimation.Fields(counter).LeafPos{length(currDeviceSeqNames),1} = [];
+               
            for k = 1:length(currDeviceSeqNames)
+           
                % beam limiting device position sequence has to be defined on
                % the first control point and has to be defined on following
-               % points only if it changes
-               currLeafPos = currControlPointElement.BeamLimitingDevicePositionSequence.(currDeviceSeqNames{k}).LeafJawPositions;          
+               % points only if it changes -> default initilation if counter > 1
+               if counter > 1
+                   tmpCollimation.Fields(counter).LeafPos{k}(:,1) = tmpCollimation.Fields(counter-1).LeafPos{k}(:,1);
+                   tmpCollimation.Fields(counter).LeafPos{k}(:,2) = tmpCollimation.Fields(counter-1).LeafPos{k}(:,2);
+               end
+
+               if isfield(currControlPointElement.BeamLimitingDevicePositionSequence,currDeviceSeqNames{k})
+                   currLeafPos = currControlPointElement.BeamLimitingDevicePositionSequence.(currDeviceSeqNames{k}).LeafJawPositions;          
+
+                   deviceIx = find(strcmp({device(:).DeviceType}, ...
+                       currControlPointElement.BeamLimitingDevicePositionSequence.(currDeviceSeqNames{k}).RTBeamLimitingDeviceType));
 
-               if (length(currLeafPos) ~= 2 * device(k).NumOfLeafs)
-                   warning(['Number of leafs/jaws does not match given number of leaf/jaw positions in control point sequence ' ...
-                            currControlPointSeqNames{j} ' on beam sequence ' BeamSeqNames{i} ' for device ' ...
-                            device(k).DeviceType '. No field shape import performed!']);
-                   return;
```

Comment:
I just changed my mind on this. I figure it is a good thing that the import at least works for the other structures..